### PR TITLE
add netgen

### PR DESCRIPTION
**Pull request by looooo**

### CODE REVIEW

File: recipes/netgen/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package:
+    name: netgen
+    version: 6.2
+
+source:
+    fn: netgen.tar.gz
+    url: https://github.com/NGSolve/netgen/archive/a1d1df91b7912a7acb9864301cb108b2e01b4882.tar.gz
```
Comment (CJ-Wright) [inline recipes/netgen/meta.yaml]: Would you mind using the standard jinja2 variables (at least for the url)?

Comment (looooo) [reply recipes/netgen/meta.yaml]: thanks, done.

**Comment by looooo** [comment]
something is wrong here:

```
-- Installing: /Users/travis/miniconda/conda-bld/netgen_1520623962641/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/Contents/Resources/Users/travis/miniconda/conda-bld/netgen_1520623962641/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.5/site-packages/netgen/__init__.py
```
but:
```
bld/netgen_1520623962641/test_tmp/run_test.py
import: 'netgen'
Traceback (most recent call last):
  File "/Users/travis/miniconda/conda-bld/netgen_1520623962641/test_tmp/run_test.py", line 2, in <module>
    import netgen
ImportError: No module named 'netgen'
```

**Comment by looooo** [comment]
@conda-forge/core 
I am using this to ensure the same python version is used for build and run. Does this work with conda-build3 or is there a better way to ensure the same python version is used.
```
-requirements
    ....
    -run
    ....
        - python {{ environ['PY_VER'] + '*' }}
```

**Comment by dougalsutherland** [comment]
@looooo Same minor version should be enforced automatically by conda (e.g. if it's built on 3.5 then it won't install on 3.6). Do you need something more than that for some reason?

File: recipes/netgen/build.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+echo "-------------------CUSTOM VARIABLES-------------"
+export NETGENDIR=$PREFIX/lib/netgen
+echo "set NETGENDIR to: $NETGENDIR"
+
+export PYTHONPATH=$SP_DIR
```
Comment (dougalsutherland) [inline recipes/netgen/build.sh]: Is this actually needed? It probably shouldn't be, though maybe netgen has a bad CMakeLists for some reason.

File: recipes/netgen/python.patch [hunk 3]
```diff
@@ -0,0 +1,28 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (dougalsutherland) [inline recipes/netgen/python.patch]: This patch isn't used in the recipe?

File: recipes/netgen/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    name: {{ name }}
+    version: {{ version }}
+
+source:
+    fn: netgen{{ version }}.tar.gz
+    url: https://github.com/NGSolve/netgen/archive/{{ commit }}.tar.gz
+    sha256: {{ sha256 }}
+    patches:
+        - python1.patch
+        - salome.patch
```
Comment (dougalsutherland) [inline recipes/netgen/meta.yaml]: What are these patches for? And have they been sent upstream?

File: recipes/netgen/build.sh (hunk 2 above)
Comment (looooo) [reply recipes/netgen/build.sh]: I don't know if it work without this. And yes netgens CMakeLists could be improved. But I am happy that it works at all.

File: recipes/netgen/python.patch (hunk 3 above)
Comment (looooo) [reply recipes/netgen/python.patch]: yes I replaced it with python1.patch. I wasn't sure if it works with the updated patch so the old patch is still there. I am going to delete this patch.

File: recipes/netgen/meta.yaml (hunk 4 above)
Comment (looooo) [reply recipes/netgen/meta.yaml]: The python1.patch includes some conda specific cmake stuff. I have already tried to apply it upstream, but it's a bit difficult. The salome-patch adds some functionality originally added by http://salome-platform.org/. I guess it's better to remove this patch and wait until the PR is merged...

**Comment by looooo** [comment]
>@looooo Same minor version should be enforced automatically by conda (e.g. if it's built on 3.5 then it won't install on 3.6). Do you need something more than that for some reason?

@dougalsutherland Thanks for the information. I added this because I thought it was the reason for a failing ci. But I will try again to run without the mentioning the vrsion.

**Comment by looooo** [comment]
@conda-forge/core I think this is now ready for the feedstock. Thanks to everyone who contributed.
@mhochsteger @dougalsutherland @CJ-Wright

**Comment by looooo** [comment]
@conda-forge/core is depending on gcc good practise? I wonder because conda-build3 will change some things in this regard.

File: recipes/netgen/python.patch (hunk 3 above)
Comment (dougalsutherland) [reply recipes/netgen/python.patch]: Note: since been changed back to use this patch.

**Comment by dougalsutherland** [comment]
Sorry for losing track of this @looooo.

Are you depending on `gcc` because it doesn't work otherwise, or just because "I need a compiler so I should use `gcc`"? If the latter, it's currently better conda-forge practice _not_ to do that and to just use the system compilers when they work (especially on Mac). I don't see any commits in the history that didn't use the `gcc` dep; did you try it without?

**Comment by looooo** [comment]
the package was previously maintained in this repo: https://github.com/FreeCAD/FreeCAD_Conda
In the beginning I used a system gcc. But I guess some dependencies are using conda-forge gcc and therefor I used it too.
I can remove conda-forge gcc dependency once conda-build3 can be used.

**Comment by dougalsutherland** [comment]
Please give it a try without the `gcc` dep. It should work fine in the conda-forge context.

**Comment by dougalsutherland** [comment]
Okay, I think this should work fine without the `gcc` dep. I'm happy to merge now; @looooo, do you think the `gcc` is necessary for some reason?

**Comment by looooo** [comment]
looks good! thanks for testing.

**Comment by dougalsutherland** [comment]
Okay, merged! Thanks @looooo.

**Pull request merged**